### PR TITLE
[Android] Default invalid imageSize values to Medium

### LINKED ISSUE #4043: [iOS-Android-Web][ImageSet Rendering] [No parity in rendering of images in imageSet for various imageSizes]
# Platform

What platform is your issue or question related to? (Delete other platforms).

- [x] Android
- [x] iOS
- [ ] UWP
- [x] JS
- [ ] .NET
- [ ] Documentation

# Author or host

Host - Teams

What version are you using? Ex: NuGet 1.0.2, or latest master, etc...

1.2.7 on iOS

# Details

When using imageSize for imageSet, there is no parity between the rendering experience on the three clients ( iOS, Android and Web). Sample JSON and screenshots are attached.

|Card Number|iOS|Android|Web|
|:-:|:-:|:-:|:-:|
|1: Large Images|drawing|drawing|Screenshot 2020-05-20 at 1 33 24 PM|
|1: Small Images|drawing|drawing|Screenshot 2020-05-20 at 1 35 28 PM|

Card JSON:
Json for card with Large images: 
[imageSetLarge.txt](https://github.com/microsoft/AdaptiveCards/files/4655476/imageSetLarge.txt)

Json for card with Small images: 
[imageSetSmall.txt](https://github.com/microsoft/AdaptiveCards/files/4655478/imageSetSmall.txt)

### PR DESCRIPTION
**Pull request by golddove**
## Related Issue
Fixes #4043 

## Description
"auto" and "stretch" are no longer valid sizes - default all invalid/empty values to "medium"

## How Verified
Manual verification w/ the following payload:
```
{
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "type": "AdaptiveCard",
    "version": "1.2",
    "body": [
        {
            "type": "ImageSet",
            "imageSize": "auto",
            "images": [
                {
                    "type": "Image",
                    "url": "https://static.thenounproject.com/png/22802-200.png"
                }
            ]
        },
        {
            "type": "ImageSet",
            "imageSize": "stretch",
            "images": [
                {
                    "type": "Image",
                    "url": "https://static.thenounproject.com/png/22802-200.png"
                }
            ]
        },
        {
            "type": "ImageSet",
            "images": [
                {
                    "type": "Image",
                    "url": "https://static.thenounproject.com/png/22802-200.png"
                }
            ]
        },
        {
            "type": "ImageSet",
            "imageSize": "",
            "images": [
                {
                    "type": "Image",
                    "url": "https://static.thenounproject.com/png/22802-200.png"
                }
            ]
        },
        {
            "type": "ImageSet",
            "images": [
                {
                    "type": "Image",
                    "url": "https://static.thenounproject.com/png/22802-200.png"
                }
            ]
        },
        {
            "type": "ImageSet",
            "imageSize": "medium",
            "images": [
                {
                    "type": "Image",
                    "url": "https://static.thenounproject.com/png/22802-200.png"
                }
            ]
        },
        {
            "type": "ImageSet",
            "imageSize": "Large",
            "images": [
                {
                    "type": "Image",
                    "url": "https://static.thenounproject.com/png/22802-200.png"
                }
            ]
        },
        {
            "type": "ImageSet",
            "imageSize": "Small",
            "images": [
                {
                    "type": "Image",
                    "url": "https://static.thenounproject.com/png/22802-200.png"
                }
            ]
        }
    ]
}
```

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/4342)

### CODE REVIEW

**Comment by paulcam206** [comment]
code looks good -- we should probably capture the test payload as a test sample :)

**Comment by shalinijoshi19** [comment]
@golddove / @jwoo-msft is defaulting to medium what we are going with ? If so, we should update our documentation here as well .. @RebeccaAnne FYI.

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ImageRenderer.java [hunk 1]
```diff
@@ -136,35 +136,24 @@ else if (imageSize == ImageSize.Large)
 
     private static void setImageSize(Context context, ImageView imageView, ImageSize imageSize, ImageSizesConfig imageSizesConfig, boolean isInImageSet) {
         imageView.setScaleType(ImageView.ScaleType.CENTER);
-        if (imageSize == ImageSize.Stretch)
+        if (imageSize != ImageSize.Small && imageSize != ImageSize.Medium && imageSize != ImageSize.Large)
```
Comment (shalinijoshi19) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ImageRenderer.java:139]: NIt: Readability & code maintenance: I would extract a quick function out of this for readability to make it clear what exactly it is that this check is doing - IsAllowedImageSizeValue - and then add a quick comment on there explaining the policy decision to default to medium.

**Comment by golddove** [comment]
Yeah, this is not ready for merge, do we have a PR label for that? This needs changes, pending inconsistency discussion.

**Comment by shalinijoshi19** [comment]
> Yeah, this is not ready for merge, do we have a PR label for that? This needs changes, pending inconsistency discussion.

No - just dont merge it. If it stays open for a while and you see a visit from the stalebot, go ahead and update with a quick comment on why it's blocked for context but other than that no special label required i think. We donth ave "auto-merge" rules in place.

**Comment by golddove** [comment]
Fixed by #4380

**Pull request closed without merging**